omod/src/main/java/org/openmrs/module/appui/fragment/controller/SessionFragmentController.java

```diff
@@ -10,11 +11,18 @@
 import org.openmrs.ui.framework.fragment.action.ObjectResult;
 import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletResponse;
+
 public class SessionFragmentController {
 	
 	public FragmentActionResult setLocation(@RequestParam("locationId") Location location,
-	                   @SpringBean("locationService") LocationService locationService, UiSessionContext context) {
+											@SpringBean("locationService") LocationService locationService,
+											UiSessionContext context, ServletContext servletContext, HttpServletResponse httpResponse) {
 		context.setSessionLocation(location);
+		httpResponse.setHeader("Set-Cookie", AppUiConstants.COOKIE_NAME_LAST_SESSION_LOCATION + "=" + location.getLocationId()
```

Comment:
> Won't this overwrite any previously set flags or values like JSESSIONID and possibly more?

Nope it won't overwrite any flags unless a header with the same name had already been set, the new value overwrites the previous one.

> Shouldn't you use the Cookie object?

Appui module uses servlet-api 2.5. But Cookie.setHttpOnly() is only supported since servlet 3.0.
So I think the easiest means to for setting the HttpOnly is to go with the setHeader() approach.

Here I'm quoting from: https://www.owasp.org/index.php/HttpOnly
> 
> Since Sun Java Enterprise Edition 6 (JEE 6), that adopted Java Servlet 3.0 technology, it's programmatically easy setting HttpOnly flag in a cookie.
> 
> In fact setHttpOnly and isHttpOnly methods are available in the Cookie interface [4], and also for session cookies (JSESSIONID) [5]:
> 
> Cookie cookie = getMyCookie("myCookieName");
> cookie.setHttpOnly(true);
>
> Moreover since JEE 6 it's also declaratively easy setting HttpOnly flag in session cookie, by applying the following configuration in the deployment descriptor WEB-INF/web.xml:
> 
> For Java Enterprise Edition versions prior to JEE 6 a common workaround is to overwrite the SET-COOKIE http response header with a session cookie value that explicitly appends the HttpOnly flag:
> 
> String sessionid = request.getSession().getId();
> // be careful overwriting: JSESSIONID may have been set with other flags
> response.setHeader("SET-COOKIE", "JSESSIONID=" + sessionid + "; HttpOnly");